src/lib/hooks/routing/clientSideSmartOrderRouter.ts

```diff
@@ -19,50 +21,44 @@ export function isSupportedChainId(chainId: ChainId | undefined): boolean {
 
 async function getQuote(
   {
-    type,
+    tradeType,
     tokenIn,
     tokenOut,
     amount: amountRaw,
   }: {
-    type: 'exactIn' | 'exactOut'
+    tradeType: TradeType
     tokenIn: { address: string; chainId: number; decimals: number; symbol?: string }
     tokenOut: { address: string; chainId: number; decimals: number; symbol?: string }
     amount: BigintIsh
   },
   router: AlphaRouter,
   config: Partial<AlphaRouterConfig>
-): Promise<{ data: GetQuoteResult; error?: unknown }> {
-  const currencyIn = new Token(tokenIn.chainId, tokenIn.address, tokenIn.decimals, tokenIn.symbol)
-  const currencyOut = new Token(tokenOut.chainId, tokenOut.address, tokenOut.decimals, tokenOut.symbol)
+): Promise<QuoteResult> {
+  if (!amountRaw) {
+    return { state: QuoteState.INITIALIZED }
```

Comment:
This will not actually initialize the client-side smart order router! You _must_ call `router.route` with dummy values to do so. Otherwise, the first call with actual value will take significantly longer. This is why widgets defaults to using a value of `1` and then returns after calling `router.route`.

Another option (to improve on widgets) is to call `router.route`, and return without awaiting, although you risk not actually initializing in this case, too.

Reply:
either way this is done, you should add a test guaranteeing that we actually initialize the SOR instead of returning early, so this isn't broken in the future.

Reply:
switched the ordering back to original and also left a comment